File: nx-maven/micronaut/m-m-app/src/main/java/com/example/mmapp/HelloController.java
package com.example.mmapp;

import com.example.mmlib.HelloService;
import io.micronaut.http.MediaType;
import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;
import io.micronaut.http.annotation.Produces;
import jakarta.inject.Inject;

@Controller("/hello")
public class HelloController {

  @Inject
  private HelloService helloService;

  @Get
  @Produces(MediaType.TEXT_PLAIN)
  public String index() {
    return helloService.greeting();
  }
}

File: nx-maven/micronaut/m-m-kt-app/src/main/kotlin/com/example/mmktapp/HelloController.kt
package com.example.mmktapp

import com.example.mmktlib.HelloService
import io.micronaut.http.MediaType
import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Get
import io.micronaut.http.annotation.Produces
import jakarta.inject.Inject

@Controller("/hello")
class HelloController(
  @Inject val helloService: HelloService,
) {
  @Get
  @Produces(MediaType.TEXT_PLAIN)
  fun index() = helloService.greeting()
}
